table/table_test.cc

```diff
@@ -7004,7 +7004,8 @@ TEST_F(ExternalTableTest, SstReaderTest) {
   std::shared_ptr<ExternalTableFactory> factory =
       std::make_shared<DummyExternalTableFactory>(
           /*support_property_block=*/false);
-  options.table_factory = NewExternalTableFactory(factory);
+  options.table_factory.reset(
+      NewExternalTableFactoryAsUniquePtr(factory).release());
```

Comment:
Late to the party, but unique_ptr can be implicitly converted to shared_ptr. It just uses an extra small allocation that is perfectly fine for when the pointer tracking is not performance critical. Probably could just have just changed the return type on NewExternalTableFactory(). And also avoid this reset+release code.